disttask/framework/scheduler/manager.go

```diff
@@ -139,11 +132,12 @@ func (m *Manager) Stop() {
 }
 
 // fetchAndHandleRunnableTasks fetches the runnable tasks from the global task table and handles them.
-func (m *Manager) fetchAndHandleRunnableTasks(ctx context.Context) {
+func (m *Manager) fetchAndHandleRunnableTasks() {
+	defer tidbutil.Recover(metrics.LabelDomain, "fetchAndHandleRunnableTasks", m.fetchAndHandleRunnableTasks, false)
```

Comment:
even if we prevent the whole runtime exiting by recover, the `fetchAndHandleRunnableTasks` is dead and no tasks will be further proceed, what should we do in this situation?

Reply:
> even if we prevent the whole runtime exiting by recover, the `fetchAndHandleRunnableTasks` is dead and no tasks will be further proceed, what should we do in this situation?

Nice catch, I will add more tests for `event loop canceled`, `panic after lock` cases.